modules/latex.php

```diff
@@ -19,88 +19,464 @@
+				'processEnvironments' => false,
+				// Elements with this class will not be processed by MathJax.
+				'ignoreClass' => 'math_ignore',
+				// Elements with this class inside of ignored elements will be processed.
+				'processClass' => 'math_process',
+			),
+			// Specify a z-index for MathMenu so it appears properly when used inside modals.
+			'MathMenu' => array(
+				'styles' => array(
+					'.MathJax_Menu' => array('z-index' => 2001),
+				),
+			),
+			// 'showMathMenu' => false,
+		);
+
+		// Users who can edit posts or pages should see math parse error details.
+		if ( current_user_can( 'edit_posts' ) || current_user_can( 'edit_pages' ) ) {
+			$config['TeX']['noErrors']    = array( 'disabled' => true );
+			$config['TeX']['noUndefined'] = array( 'disabled' => true );
+		}
+
+		return $config;
 	}
-	if ( preg_match( '/.+(&s=[0-9-]{1,2}).*/i', $latex, $s_matches ) ) {
-		$s = (int) substr( $s_matches[1], 3 );
-		$latex = str_replace( $s_matches[1], '', $latex );
+
+	/**
+	 * Output in-line MathJax configuration.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @link http://docs.mathjax.org/en/latest/configuration.html#using-in-line-configuration-options
+	 * @link https://github.com/mathjax/MathJax-website/commit/df7f655623b1a6fc0bc637d9f4b47cb3461aa118
+	 */
+	public function latex_mathjax_config_output() {
+		/**
+		 * Allow MathJax configuration to be overridden.
+		 *
+		 * There are many more configuration options available than are outlined
+		 * here. Please see the MathJax documentation for a full list.
+		 *
+		 * @link http://docs.mathjax.org/en/latest/index.html
+		 *
+		 * @since 3.8.0
+		 *
+		 * @param array $config {
+		 *     In-line MathJax configuration options.
+		 *
+		 *     @type array $config     Use a pre-defined configuration file.
+		 *     @type array $jax        Specify input and output methods for MathJax.
+		 *     @type array $extensions Specify which MathJax extensions to load.
+		 *     @type array $TeX        Define Macros, specify which TeX extensions to load, and more.
+		 *     @type array $tex2jax    Define math delimiters and other options.
+		 *     @type array $MathMenu   Control the contextual menu that is available on mathematics that are typeset by MathJax.
+		 * }
+		 */
+		$config = apply_filters( 'latex_mathjax_config', $this->latex_mathjax_config() );
+
+		?>
+		<script type="text/x-mathjax-config">
+			MathJax.Hub.Register.StartupHook( 'MathMenu Ready', function() {
+				MathJax.Menu.BGSTYLE['z-index'] = 2000;
+			} );
+			MathJax.Hub.Config( <?php echo wp_json_encode( $config ); ?> );
+		</script>
+	<?php }
+
+	/**
+	 * Output MathJax.js script tag on the page if it hasn't already been done.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @link http://docs.mathjax.org/en/latest/configuration.html#using-in-line-configuration-options
+	 * @link https://github.com/mathjax/MathJax-website/commit/df7f655623b1a6fc0bc637d9f4b47cb3461aa118
+	 *
+	 * @return boolean
+	 */
+	public function output_mathjax_script() {
```

Comment:
This will load the script once per page, which is good.  When an Infinite Scroll AJAX request loads the next batch of posts, though, it will output the script again (and again for each AJAX request).

There's a better way to do this.  As long as you always add scripts to the page using `wp_enqueue_script()`, Infinite Scroll will detect what scripts are new and add only those.  Check out theme-tools/responsive-videos.php, for example, which calls `wp_enqueue_script()` to let WP and Infinite Scroll handle everything.